axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolServiceImpl.java

```diff
@@ -478,4 +475,21 @@ public boolean getEditAuthorization(Move move) throws AxelorException {
 
     return result;
   }
+
+  @Override
+  public String filterJournalPartnerCompatibleType(Move move) {
+    StringBuilder compatiblePartnerDomain = new StringBuilder("self.id IN (");
+    Journal journal = move.getJournal();
+    String[] compatiblePartnerTypeSelect = journal.getCompatiblePartnerTypeSelect().split(",");
+    Set<Long> compatiblePartnerIds = new HashSet<>();
+    for (String compatiblePartnerType : compatiblePartnerTypeSelect) {
+      compatiblePartnerIds.addAll(partnerService.getPartnerIdsByType(compatiblePartnerType));
+    }
```

Comment:
I think this part can be moved to `JournalService` in case a similar process is needed for another objet than `Move`